### DISCUSSION TITLE
Save query results for bug report

### DISCUSSION THREAD

**Discussion by xingus** [datalust/seq-tickets#1568] [Help] [answered]
Is there a way to save a set of displayed events to reference in a bug report? Say I have a query which selects 30ish events that demonstrate a bug. 

I could export to csv or json. Neither is very usable, especially compared to Seq's great UI.

I could past the query in the bug report, but if it's not high priority, the events might not be retained long enough, so I have to expand each event and permalink it. This is tedious for more than a few events. is there a way to "permasave" all displayed entries?

Something more like a signal that saves the current selected events, and could be linked from a bug report with a few clicks would be great. Is there a way to do this?

**Comment by liammclennan** [comment, marked as answer]
Use [seqcli search](https://docs.datalust.co/docs/command-line-client#search) with the `--json` option to export the events in [CLEF format](https://clef-json.org/). 

```shell
seqcli search -c 10000000 --json -s https://your-server -a your-api-key -f "Application = 'App 1'" > saved-events.clef
```

Should you need to analyse those events later you can use [seqcli ingest](https://docs.datalust.co/docs/command-line-client#ingest) to load them back into Seq:

```shell
seqcli ingest --json -i saved-events.clef -s https://some-other-server -a anotherapikey -p BugReport=9567
```

**Reply by xingus** [reply]
Thanks Liam! Not as slick as I had hoped, but does the job. 

I was able to do the export from the seq events web page. In addition to the query, I had to set the start date to get events more than 24 hours old.
